### PR TITLE
Swap gce-conformance on sig-release-master-blocking from kubetest to kubetest2

### PR DESCRIPTION
**Pull request by amwat**
First of many! No bootstrap, no scenarios, no kubetest :) 

xref: https://github.com/kubernetes/enhancements/issues/2464

The kubetest2 gce-conformance job has been stable for a while:
https://testgrid.k8s.io/conformance-all#Conformance%20-%20GCE%20-%20master%20-%20kubetest2

the old job will still be visible at: https://testgrid.k8s.io/conformance-all#Conformance%20-%20GCE%20-%20master
will not be turned down yet

Only targetting master, older release branches will still use kubetest based jobs.

/cc @BenTheElder @spiffxp 
/cc @kubernetes/ci-signal 
/cc @kubernetes/sig-release

### CODE REVIEW

**Review by spiffxp — COMMENTED** [review]
/area conformance
/cc @dims @johnbelamaric 
as FYI

I compared results of latest runs
- [latest-kubetest2](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-kubernetes-gce-conformance-latest-kubetest2/1368978837181829120)
- [latest](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-kubernetes-gce-conformance-latest/1368976318514860032)

The difference in number of tests passed is due to the `e2e.go` suite having 13 steps, where kubetest2's (unamed) suite has 4 steps

Comparing durations on testgrid:
- [latest-kubetest2](https://testgrid.k8s.io/conformance-gce#Conformance%20-%20GCE%20-%20master%20-%20kubetest2&graph-metrics=test-duration-minutes&width=5)
- [latest](https://testgrid.k8s.io/conformance-gce#Conformance%20-%20GCE%20-%20master&graph-metrics=test-duration-minutes&width=5)

`Overall` is about 10min longer for kubetest2, but `Test` is about 10min shortee for kubetest2.  Turns out this is due to the fact that there is a `kubetest2.Build` stage.

The kubetest2 job hasn't been passing for a full two weeks, but I see nothing in its history to indicate that it's providing lower quality signal than the job it's replacing.

So, suggestions for followup:
- kubetest2 job should download pre-built binaries from `k8s-release-dev` bucket
- kubetest2 should give its "meta" suite a name, maybe `kubetest2`

/approve
/lgtm
/hold
for comment, I'm gonna go ping release team

**Comment by spiffxp** [comment]
/assign @thejoycekung @palnabarun 
Release team: CI Signal lead, Release lead
/assign @hasheddan @jeremyrickard 
SIG Release TLs

I'll remove `/hold` If no objections by SIG Testing meeting tomorrow at 10am PT, or +1's from at least one of each `/assign` line

**Review by hasheddan — APPROVED** [review]
/lgtm

Thanks @amwat!

**Comment by amwat** [comment]
+1 ack on the differences

- Adding support for GCE extract (https://github.com/kubernetes-sigs/kubetest2/issues/106) (although master job should ideally be building from master? 🤔 )
- testsuite name was added in https://github.com/kubernetes-sigs/kubetest2/pull/93 and it shows up as such in testgrid,  need to figure out why it's not showing up in the junit lens

**Comment by BenTheElder** [comment]
> Adding support for GCE extract (kubernetes-sigs/kubetest2#106) (although master job should ideally be building from master? 🤔 )

No, the vast majority of jobs targeting any branch use the existing uploads of this. It adds a little latency (the time it takes us to `make quick-release` + upload) but saves us a lot of compute, we have a continuous build to consume.

(I need to fix this for KIND, but it was not subsuming an existing job, and for some time we were intentionally using different builds in some jobs).

**Comment by chizhg** [comment]
> * testsuite name was added in [kubernetes-sigs/kubetest2#93](https://github.com/kubernetes-sigs/kubetest2/pull/93) and it shows up as such in testgrid,  need to figure out why it's not showing up in the junit lens

Do you have a spyglass link for this? I checked one of our test runs and the `kubetest2` testsuites looks good in the junit lens.

**Comment by thejoycekung** [comment]
/lgtm

The job seems pretty stable and I think it's awesome we're moving away from bootstrap. Thanks y'all!

**Review by thejoycekung — APPROVED** [review]
/lgtm

The job looks stable, and I think it's super awesome we're moving away from bootstrap. Thanks y'all!

**Comment by amwat** [comment]
> Do you have a spyglass link for this? I checked one of our test runs and the kubetest2 testsuites looks good in the junit lens.

tracked in https://github.com/kubernetes-sigs/kubetest2/issues/107

**Comment by spiffxp** [comment]
/hold cancel
Thanks for the review all!

**Pull request merged**
